docs/topics/opt-in-requirements.md

```diff
@@ -271,21 +393,56 @@ fun getTime(): Time {}
+
+In such cases, you want to require users to opt in to your API before they can extend it further.
+For example, by inheriting from the API or declaring implementations for abstract functions. Using the [`@SubclassOptInRequired`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-subclass-opt-in-required/)
+annotation, you can enforce this behavior for [open](inheritance.md) or [abstract classes](classes.md#abstract-classes) and [non-functional interfaces](interfaces.md).
+
+To add the opt-in requirement to an API element, use the [`@SubclassOptInRequired`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-subclass-opt-in-required/)
+annotation with a reference to the annotation class:
+
+```kotlin
+@RequiresOptIn(
+ level = RequiresOptIn.Level.WARNING,
+ message = "Interfaces in this library are experimental"
+)
+annotation class UnstableApi()
+
+@SubclassOptInRequired(UnstableApi::class)
+// An interface requiring opt-in to extend
+interface CoreLibraryApi 
+```
+
+Note that when you use the `@SubclassOptInRequired` annotation to require opt-in, the requirement is not propagated to
+any [inner or nested classes](nested-classes.md).
+
+For a real-world example of how to use the `@SubclassOptInRequired` annotation in your API, check out the
+[`SharedFlow`](https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines.flow/-shared-flow/)
+interface in the `kotlinx.coroutines` library.
 
 ## Opt-in requirements for pre-stable APIs
 
 If you use opt-in requirements for features that are not stable yet, carefully handle the API graduation to avoid 
-breaking the client code.
+breaking client code.
 
-Once your pre-stable API graduates and is released in a stable state, remove its opt-in requirement annotations from declarations.
-The clients will be able to use them without restriction. However, you should leave the annotation classes in modules so that 
-the existing client code remains compatible.
+Once your pre-stable API graduates and is released in a stable state, remove any opt-in requirement annotations from 
+your declarations. The clients can then use them without restriction. However, you should leave the annotation classes 
+in modules so that the existing client code remains compatible.
 
-To let the API users update their modules accordingly (remove the annotations 
-from their code and recompile), mark the annotations as [`@Deprecated`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-deprecated/)
-and provide the explanation in the deprecation message.
+To encourage API users to update their modules by removing any annotations from their code and recompiling, mark the annotations
+as [`@Deprecated`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-deprecated/) and provide an explanation in the deprecation message.
 
 ```kotlin
 @Deprecated("This opt-in requirement is not used anymore. Remove its usages from your code.")
```

Comment:
Could you please add +1 enter to the bottom of the page?

Reply:
Sure. I'm curious though. Why? :)

Reply:
I'm not sure why JB docs follow this convention, I've seen that most of our docs follow it.  Personally, my IDE always sends me an error if I don't put a new line at the end of a text file because of posix compliance